crates/core/src/syn/parser/stmt/alter.rs

```diff
@@ -79,25 +75,132 @@ impl Parser<'_> {
 						_ => unexpected!(self, peek, "`NORMAL`, `RELATION`, or `ANY`"),
 					}
 				}
-				t!("SCHEMALESS") => {
+				t!("SCHEMALESS") if !drop => {
 					self.pop_peek();
 					res.full = Some(false);
 				}
-				t!("SCHEMAFULL") => {
+				t!("SCHEMAFULL") if !drop => {
 					self.pop_peek();
 					res.full = Some(true);
 				}
-				t!("PERMISSIONS") => {
+				t!("PERMISSIONS") if !drop => {
 					self.pop_peek();
 					res.permissions = Some(ctx.run(|ctx| self.parse_permission(ctx, false)).await?);
 				}
 				t!("CHANGEFEED") => {
 					self.pop_peek();
-					if self.eat(t!("NONE")) {
-						res.changefeed = Some(None);
+					res.changefeed = Some(if drop {
+						None
 					} else {
-						res.changefeed = Some(Some(self.parse_changefeed()?));
+						Some(self.parse_changefeed()?)
+					})
+				}
+				_ if drop => {
+					unexpected!(self, peek, "`COMMENT` or `CHANGEFEED`")
+				}
```

Comment:
Could you just add `t!("DROP")` to the match and then properly parse the keywords which are allowed after instead of adding this drop flag which makes the match more less readable and complicates flow.